tests/MongoDB.Bson.Tests/Jira/CSharp2643Tests.cs

```diff
@@ -0,0 +1,51 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using FluentAssertions;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.Serialization;
+using Xunit;
+
+namespace MongoDB.Bson.Tests.Jira
+{
+    public class CSharp2643Tests
```

Comment:
We don't normally like using test classes named after the JIRA ticket but I can't think of a reasonable place to put this test.

Reply:
I agree with @DmitryLukyanov on this one, especially given that changes were made in ImmutableTypeClassMapConvention.cs